### PR TITLE
Remove np.asarray in formatting.py

### PR DESCRIPTION
**Pull request by dcherian**

### CODE REVIEW

File: xarray/core/pycompat.py [hunk 1]
```diff
@@ -101,3 +101,37 @@ def is_chunked_array(x) -> bool:
 
 def is_0d_dask_array(x):
     return is_duck_dask_array(x) and is_scalar(x)
+
+
+def to_numpy(data):
```
Comment (headtr1ck) [inline xarray/core/pycompat.py:106]:
```suggestion
def to_numpy(data) -> np.ndarray:
```
Not sure what data should be... maybe simply Any?

File: xarray/core/pycompat.py [hunk 2]
```diff
@@ -101,3 +101,37 @@ def is_chunked_array(x) -> bool:
...
+    if isinstance(data, array_type("pint")):
+        data = data.magnitude
+    if isinstance(data, array_type("sparse")):
+        data = data.todense()
+    data = np.asarray(data)
+
+    return data
+
+
+def to_duck_array(data):
```
Comment (headtr1ck) [inline xarray/core/pycompat.py:129]: Some typing would be nice, but again no idea what the return type is. Do we have a duck array protocol?

I think for internal use a protocol is nice, for external return types we should simply return Any.

**Review by headtr1ck — APPROVED** [review]

**Comment by dcherian** [comment]
This module has no typing so I'd like to get this in and sort that out later.

**Comment by headtr1ck** [comment]
> This module has no typing so I'd like to get this in and sort that out later.

Fine by me :)

**Review by andersy005 — APPROVED** [review]

**Pull request merged**